### PR TITLE
📧 Add email decorator

### PR DESCRIPTION
**Pull request by bdsoha**
Add a decorator for the `EmailOperator`.

```python
@task.email
def send():
    ...
```

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/decorators/email.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+from typing import Any, Callable, Collection, Mapping, Sequence
+
+from airflow.decorators.base import DecoratedOperator, TaskDecorator, task_decorator_factory
+from airflow.operators.email import EmailOperator
+from airflow.utils.context import Context, context_merge
+from airflow.utils.operator_helpers import determine_kwargs
+from airflow.utils.types import NOTSET
+
+
+class _EmailDecoratedOperator(DecoratedOperator, EmailOperator):
```
Comment (eladkal) [inline airflow/decorators/email.py]: Should this be here?
I think it should be in smtp provider
https://github.com/apache/airflow/blob/main/airflow/providers/smtp/operators/smtp.py

Comment (bdsoha) [reply airflow/decorators/email.py]: Just made the move!

**Comment by bdsoha** [comment]
@eladkal Mind having a look?
Thanks.

**Comment by uranusjr** [comment]
Can you add an example DAG for this?

**Comment by bdsoha** [comment]
@eladkal Hey, any chance you can review this?
Thanks!

**Comment by potiuk** [comment]
You have now conflicts and you need to rebase it first.

**Pull request closed without merging**